src/application.cr

```diff
@@ -9,18 +9,24 @@ module Artanis
     include DSL
     include Render
 
-    getter :context, :params
+    getter :context
 
     # TODO: parse query string and populate @params
     # TODO: parse request body and populate @params (?)
     def initialize(@context : HTTP::Server::Context)
       @params = {} of String => String
+      @parsed_body = false
     end
 
     def request
       context.request
     end
 
+    def params
+      parse_body_params
+      @params
+    end
```

Comment:
I'd like to avoid boolean, so maybe `@params` could become a nilable?

```crystal
def params
  @params ||= parse_body_params
end

def parse_body_params
  params = {} of String => String

  if request.headers["Content-Type"]? == "application/x-www-form-urlencoded"
    if body = request.body
      HTTP::Params.parse(body.gets_to_end) { |key, value| @params[key] = value }
    end
  end

  params
end
```